### PR TITLE
add DefaultDataSubCategoryType

### PR DESCRIPTION
**Pull request by MadDataScience**
## Related Issues

- [T-13720](https://transcend.height.app/T-13720) Schema classifier service
- [T-14003](https://transcend.height.app/T-14003) Default data subcategories
- [T-13996](https://transcend.height.app/T-13996) Enumerate categories and test them
- closes T-14018

## Security Implications

_[none]_

## System Availability

_[none]_

### CODE REVIEW

**Review by anotherminh — DISMISSED** [review]
actually a really cool way to hook up the existing identifier types 🤯

File: src/subcategories/index.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { makeEnum } from '@transcend-io/type-utils';
+
+import { FinancialSubCategory } from './financial';
+import { CommunicationIdentifierType } from '../identifier';
+
+export const DefaultDataSubCategoryType = makeEnum({
+  ...CommunicationIdentifierType,
```
Comment (LuGary) [inline src/subcategories/index.ts:7]: I would create a new enum just for communication/contact subcategories. It seems like `CommunicationIdentifierType` is used by `IdentifierType` which we don't want to extend if we just wanted to add a new contact subcategory.

Comment (MadDataScience) [reply src/subcategories/index.ts:7]: I'm okay with that. It does mean that there will be some redundancy, but maybe that's okay if we're considering identifiers and datasubcategories to be essentially different and only coincidentally having some overlapping values.

**Review by LuGary — APPROVED** [review]
👍 

Given this is a public repo, do we want to have links to Height in the TODOs? These TODO links aren't sensitive but maybe some others will be, not sure

**Comment by MadDataScience** [comment]
> Given this is a public repo, do we want to have links to Height in the TODOs? These TODO links aren't sensitive but maybe some others will be, not sure

I wasn't sure about that either. @michaelfarrell76 or @anotherminh , what do you think? Should I also remove those links from the **Related Issues** section above?

**Comment by anotherminh** [comment]
I'm ok with leaving them honestly. If we're serious about making these open-source then i think the associated issues/tasks should be open-sourced too (i know they're not right now bc Height requires auth but the spirit of it is that it should be OK for the public to discover i think)

**Review by anotherminh — APPROVED** [review]

**Comment by michaelfarrell76** [comment]
@MadDataScience @anotherminh @LuGary good q. i dont have a strong opinion. i agree that it would be nice if the tickets were viewable, but i also feel like its much easier to track in height and itll get done faster if we track it there alongside everything else.

**Pull request merged**